lib/galaxy/tools/__init__.py

```diff
@@ -2541,7 +2541,7 @@ def iterate(group_inputs, inputs, state_inputs, other_values=None):
             other_values = ExpressionContext( state_inputs, other_values )
             for input_index, input in enumerate( inputs.itervalues() ):
                 # create model dictionary
-                tool_dict = input.to_dict(trans)
+                tool_dict = input.to_dict(trans, other_values=other_values)
```

Comment:
@guerler Thanks for letting us know - this change didn't break any of the unit tests.  Unfortunately, eliminating this change makes the other changes irrelevant, so I'm hopeful there is a way to pass the tool inputs around to the many methods that are involved in ultimately building the tool model.  

It seems strange to me that the various to_dict() functions that are used to json-ify tool input parameters cannot *all* have access to the entire set of inputs while some can.  For example, I found the comment near line #1037 in basic.py (near the change below) strange:

        # Get options, value.
        options = []
        try:
            options = self.get_options( trans, other_values )
        except AssertionError:
            # we dont/cant set other_values (the {} above), so params that require other params to be filled will error:
            #       required dependency in filter_options
            #       associated DataToolParam in get_column_list
            pass

It seems that it is known / acceptable behavior for some dynamically generated parameters to throw exceptions.  Is there a way we can fix this so that grouping parameters, along with dynamic select lists behave as expected?